plutus-tx-plugin/test/ByteStringLiterals/Spec.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE BlockArguments    #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeApplications  #-}
+
+module ByteStringLiterals.Spec (tests) where
+
+import Data.ByteString (ByteString)
+import Data.Char (chr)
+import Data.Foldable (for_)
+import Data.String (fromString)
+import PlutusPrelude (display)
+import PlutusTx (CompiledCode, getPlcNoAnn)
+import PlutusTx.Builtins (BuiltinByteString, fromBuiltin)
+import PlutusTx.Builtins.HasOpaque (stringToBuiltinByteString)
+import PlutusTx.TH (compile)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase, (@?=))
+import UntypedPlutusCore (Program (_progTerm))
+
+tests :: TestTree
+tests =
+  testGroup
+    "ByteStringLiterals"
+    [ test_FromString
+    , testGroup
+        "Compile BuiltinByteString Literal"
+        [ test_CompileBuiltinByteStringLiteral_IsString
+        , test_CompileBuiltinByteStringLiteral_StringToBuiltinByteString
+        ]
+    ]
+
+test_FromString :: TestTree
+test_FromString = testCase "fromString" do
+  for_ [0x00 .. 0xFF] \(i :: Int) -> do
+    let s :: String = [chr i]
+    fromBuiltin (fromString @BuiltinByteString s) @?= fromString @ByteString s
+
+test_CompileBuiltinByteStringLiteral_IsString :: TestTree
+test_CompileBuiltinByteStringLiteral_IsString =
+  testCase "OverloadedStrings" do
+    display (_progTerm (getPlcNoAnn compiledLiteral)) @?= expectedUplc
+  where
+    compiledLiteral :: CompiledCode BuiltinByteString =
+      $$( compile
+            [||
+            "\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\
```

Comment:
This is probably going to sound ridiculous, but can you add another such tests, but with bytes being in the opposite order? Just to make sure that the tests pass not because we got lucky with ordering of bytes. I'm perhaps saying nonsense, but it'll also perhaps be easier to add another couple of tests that bring me to my senses.

Reply:
I see your point! I had similar thought: what if the way bytes are ordered influences how they are decoded? While I don't expect this to be the case, adding a test case is cheap and easy so I'll simply do it.

Reply:
Done.